core/src/main/scala/kafka/controller/KafkaController.scala

```diff
@@ -1691,6 +1691,144 @@ class KafkaController(val config: KafkaConfig, zkUtils: ZkUtils, time: Time, met
         triggerControllerMove()
     }
   }
+
+  case object BrokerChangeV2 extends ControllerEvent {
+    override def state: ControllerState = ControllerState.BrokerChange
+
+    override def process(): Unit = ???
+  }
+
+  case object TopicChangeV2 extends ControllerEvent {
+    override def state: ControllerState = ControllerState.TopicChange
+
+    override def process(): Unit = ???
+  }
+
+  case object LogDirEventNotificationV2 extends ControllerEvent {
```

Comment:
Hmm, do we need those wrappers? Could be just change ControllerEvent directly?

Reply:
Anything with a V2 suffix is just temporary. It's there to make the diff easier to read. Once we agree on the changes, I'll remove the suffix and make the new code replace the old.